kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -93,7 +97,12 @@ public KafkaProxyInitializer(FilterChainFactory filterChainFactory,
 
     @Override
     public void initChannel(Channel ch) {
-
+        if (!ch.hasAttr(SESSION_ID_ATTRIBUTE_KEY)) {
+            String sessionId = MobyNamesGenerator.getRandomName() + "_" + ch.id().asShortText();
+            MDC.put(SESSION_ID_ATTRIBUTE_KEY.name(), sessionId);
```

Comment:
We talked about this https://kroxylicious.slack.com/archives/C04V1K6EAKZ/p1753185341911749?thread_ts=1753174375.102959&cid=C04V1K6EAKZ